### ISSUE TITLE
Update to Tycho 2.x

### ISSUE THREAD

**Issue by cdietrich** [eclipse/xtext#1816]
Note: Tycho 2.0.0 needs Java 11 to run. am not sure about 8 as target and if the toolchain can ensure that we dont use wrong api

knows problems

```
On Tue, Jul 28, 2020 at 2:51 PM Christian Dietrich <christian.dietrich@itemis.de> wrote:

    in Xtext we get

    [ERROR] Failed to resolve target definition /home/default/xtext-eclipse/releng/org.eclipse.xtext.target/org.eclipse.xtext.target-latest.target: The attribute 'includeSource' must be the same for all locations -> [Help 1]

    https://github.com/eclipse/xtext-eclipse/tree/cd_tycho200test

    is this new/intended?

Yes, see https://bugs.eclipse.org/bugs/show_bug.cgi?id=550169 , the includeSource flag is now honored by default (instead of ignored in 1.x).
Best is to have the same value for all includeSources in your .target file, alternatively, you can configure the target-platform-configuration `targetDefinitionIncludeSource` setting to "ignore" so that the includeSources settings will be ignored and build will continue.
```

**Comment by cdietrich** [comment]
https://bugs.eclipse.org/bugs/show_bug.cgi?id=565603 also is fixed now in the latest TYCHO 2.0.0-SNAPSHOT

**Comment by cdietrich** [comment]
we build latest with new tycho. for "oxygen" and others we keep 1.7 as these need java 8
